server/game/cards/02_SHD/upgrades/VambraceFlamethrower.ts

```diff
@@ -0,0 +1,31 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { UpgradeCard } from '../../../core/card/UpgradeCard';
+import { Location, RelativePlayer, Trait, WildcardCardType } from '../../../core/Constants';
+import { Card } from '../../../core/card/Card';
+
+export default class VambraceFlamethrower extends UpgradeCard {
+    protected override getImplementationId() {
+        return {
+            id: '6471336466',
+            internalName: 'vambrace-flamethrower',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.setAttachCondition((card: Card) => !card.hasSomeTrait(Trait.Vehicle));
+
+        this.addGainOnAttackAbilityTargetingAttached({
+            title: 'Deal 3 damage divided as you choose among enemy ground units',
+            immediateEffect: AbilityHelper.immediateEffects.distributeDamageAmong({
+                amountToDistribute: 3,
+                optional: true,
+                canChooseNoTargets: false,
```

Comment:
Extending the above comment re: the optional parameter. To help people catch this in the future, can you go to GameSystem.ts and change the TODO comment on line 17 to be an @deprecated so that people will see it crossed out:

```typescript
/** @deprecated TODO: evaluate whether to remove this */ 
optional?: boolean;
```